lib/curl/curl.nit

```diff
@@ -122,71 +122,156 @@ class CurlHTTPRequest
 	# instead of a TCP connection and DNS hostname resolution.
 	var unix_socket_path: nullable String is writable
 
+	# The HTTP method.
+	# Must be a capitalized string with request name complying with RFC7231
+	# Default is GET
+	var method: String = "GET" is optional, writable
+
 	# Execute HTTP request
 	#
 	# By default, the response body is returned in an instance of `CurlResponse`.
 	# This behavior can be customized by setting a custom `delegate`.
 	fun execute: CurlResponse
 	do
+		# Reset libcurl parameters as the lib is shared and options
+		# might affect requests from one another.
+		self.curl.native = new NativeCurl.easy_init
 		if not self.curl.is_ok then return answer_failure(0, "Curl instance is not correctly initialized")
 
 		var success_response = new CurlResponseSuccess
 		var callback_receiver: CurlCallbacks = success_response
-		if self.delegate != null then callback_receiver = self.delegate.as(not null)
+		var err : CURLCode
+
+		# Prepare request
+		err = prepare_request(callback_receiver)
+		if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+
+		# Perform request
+		var err_resp = perform
+		if err_resp != null then return err_resp
+
+		var st_code = self.curl.native.easy_getinfo_long(new CURLInfoLong.response_code)
+		if not st_code == null then success_response.status_code = st_code
+
+		self.curl.native.easy_clean
+
+		return success_response
+	end
+
+	# Internal function that sets cURL options and request' parameters
+	private fun prepare_request(callback_receiver : CurlCallbacks) : CURLCode
```

Comment:
To respect the english (and Nit) style, there should be a signle space only after `:`, not before.